File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i=0;
        int j=0;
        int n = nums.size();
        int calc_sum=0;
        int ans=INT_MAX;
        while(j<n) {
            calc_sum+=nums[j];
            while(i<=j&& calc_sum>=target) {
                ans = min(ans,j-i+1);
                calc_sum-=nums[i];
                i++;
            }
            j++;

        }
        return ans==INT_MAX?0:ans;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        answer = float("inf")
        running_sum = 0
        l = 0
        for r in range(len(nums)):
            running_sum+=nums[r]
            while running_sum>=target:
                answer = min(answer,r-l+1)
                running_sum-=nums[l]
                l+=1
        return 0 if answer>=float('inf') else answer
